### DISCUSSION TITLE
Is it possible to fetch data from external API?

### DISCUSSION THREAD

**Discussion by SNPR** [directus/directus#3494] [Archive]
Hi! Is it possible to implement this:
Then I create a new (e.g. string) field for my collection I click Interface -> 
![image](https://user-images.githubusercontent.com/22043414/102874961-015a1a80-4465-11eb-8b97-d29ce37c2b88.png)

and in this list of items I choose my own component (e.g. "Select").

And then I create a new item for my collection I see this Select and available options in this select I fetch from external API.

**Comment by benhaynes** [comment]
As of now, to fetch data from an external API you would need to write some logic in a custom interface or hook. We're discussing how this could be "standardized" in the future for no-code implementation.

**Comment by jairo-fernandezr** [comment]
Hi @benhaynes, is there any progress on the possible standard ways to fetch/update data using an external API ?
Your comments are very appreciated, thanks.

**Reply by benhaynes** [reply]
Yes, you can use the Flows feature that we released a few weeks ago. It's still an MVP, but we're working to clean up the dev experience now.

https://docs.directus.io/configuration/flows/#flows

**Reply by add-n2x** [reply]
@benhaynes how can this be achieved using the Flows feature? The docs could not enlighten me yet.

**Comment by casualmatt** [comment]
I just post some days a go a full Feature Requests, with an example on how to integrate external data in directus, maybe you have feedback to improve my ideas.

https://github.com/directus/directus/discussions/17504

**Comment by add-n2x** [comment]
Is there any news on this? I've tried playing around with building custom extensions, but no luck yet, as I'm not sure if I'm heading in the right direction.

Also compare [this feature request](https://github.com/directus/directus/discussions/21049#discussioncomment-9244149). 

Unfortunately @casualmatt's [feature request got closed](https://github.com/directus/directus/discussions/17504) already.
